torchqdynamics/types.py

```diff
@@ -12,25 +12,37 @@
 # type for objects convertible to a torch tensor using `torch.as_tensor`
 TensorLike = Union[List, np.ndarray, torch.Tensor]
 
-# type for objects convertible to a torch tensor using `torchqdynamics.to_tensor`
-QTensorLike = Union[TensorLike, Qobj]
+# type for objects convertible to a torch tensor using `to_tensor`
+OperatorLike = Union[TensorLike, Qobj]
 
+# type for objects convertible to a `TimeDependentOperator` using `time_dependent_to_tensor`
+TimeDependentOperatorLike = Union[OperatorLike, Callable[[float], OperatorLike]]
 
-def to_torch(x: QTensorLike) -> torch.Tensor:
-    """Convert a `QTensorLike` object to a PyTorch tensor.
+
+def to_tensor(x: Optional[Union[OperatorLike, List[OperatorLike]]]) -> torch.Tensor:
+    """Convert a `OperatorLike` object or a list of `OperatorLike` object to a PyTorch
+    tensor.
 
     Args:
-        x: QuTiP quantum object or NumPy array or Python list or PyTorch tensor.
+        x: QuTiP quantum object or NumPy array or Python list or PyTorch tensor or list
+           of these types. If `None` or empty list, returns an empty tensor of size (0).
 
     Returns:
         PyTorch tensor.
     """
+    if x is None:
+        return torch.tensor([])
+    if isinstance(x, list):
+        if len(x) == 0:
+            return torch.tensor([])
```

Comment:
I believe this is redundent with the line below, since `torch.stack([])` should return an empty tensor.

Reply:
Unfortunately:
```python
RuntimeError: stack expects a non-empty TensorList
```